core/src/test/java/org/elasticsearch/cluster/DiskUsageTests.java

```diff
@@ -184,12 +181,12 @@ public void testFillDiskUsageSomeInvalidValues() {
                 new FsInfo.Path("/most", "/dev/sdc", 300, 290, 280),
         };
         FsInfo.Path[] node2FSInfo = new FsInfo.Path[] {
-                new FsInfo.Path("/least_most", "/dev/sda", -2, -1, -1),
+                new FsInfo.Path("/least_most", "/dev/sda", -1, -1, -1),
         };
 
         FsInfo.Path[] node3FSInfo =  new FsInfo.Path[] {
                 new FsInfo.Path("/most", "/dev/sda", 100, 90, 70),
-                new FsInfo.Path("/least", "/dev/sda", 10, -8, 0),
+                new FsInfo.Path("/least", "/dev/sda", 10, -1, 0),
```

Comment:
@jasontedor this is the test I am unsure about. It failed when I added code to reject negative bytes values (expect the special -1 value) because this test explicitly checked that negative values for disk stats were accepted but the disk is skipped. From looking at [FSProbe#L143](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/monitor/fs/FsProbe.java#L143) which is where we gather the disk statistics it seems that we can't actually get negative values here and instead any value reported by the OS as negative are interpreted by FSProbe as an overflow and are converted to `Long.MAX_VALUE`. 

I changed this test to use the special -1 value instead and check the disks are still skipped but maybe we actually don't need this test at all?

Reply:
I agree, the test was doing unrealistic things so I am fine with the change you made here. Let's keep whether or not the test is needed as a separate issue. There's probably other cleanup to do here, like adding asserts in FsInfo.Path about not seeing negative values.